unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/WearableGridController.cs

```diff
@@ -138,10 +138,20 @@ public void UnEquip(string wearableId)
             view.SetWearable(wearableGridModel with { IsEquipped = false });
         }
 
+        public void UpdateBodyShapeCompatibility(string bodyShapeId)
+        {
+            foreach ((string wearableId, WearableGridItemModel model) in currentWearables)
+            {
+                if (!wearablesCatalogService.WearablesCatalog.TryGetValue(wearableId, out WearableItem wearable)) continue;
+                bool isCompatibleWithBodyShape = IsCompatibleWithBodyShape(bodyShapeId, wearable);
+                view.SetWearable(model with { IsCompatibleWithBodyShape = isCompatibleWithBodyShape });
```

Comment:
You allocate a whole new instance of `WearableGridItemModel` on the heap with every `with`. Considering you do it in a loop potentially over a big number of `wearables` it may lead to an undesired `GC` pressure. I would recommend refraining from `record`  and converting `WearableGridItemModel` to a `struct` instead. Then you can just set this flag with no extra memory overhead.

Reply:
This change brings many issues in the implementation. The most important one is the data inconsistency when modifying the instances in `currentWearables` dictionary (ie: equip, unequip, updateBodyShapeCompatibility). I can set a field of the struct and then update the view, but i also need to update the dictionary with the new state, which introduces invalid operation exceptions inside a loop. In general, since the object reference is not being modified, it may lead to different data inconsistency issues.
Apart from that, im not too worried about GC, since `currentWearables` may hold at most 15 elements, plus `UpdateBodyShapeCompatibility` is not frequently called either.

Reply:
I've removed the allocation. Instead updates the model's properties. I keep the `record` since changing to struct brings many implementation issues. The most important one is the possible data inconsistency of the `currentWearables` dictionary every time the state changes, because its not updated by reference. Also introduces new invalid operations exceptions when trying to update the model instance in the dictionary insde a loop.